CorsixTH/Lua/dialogs/build_room.lua

```diff
@@ -102,16 +102,19 @@ function UIBuildRoom:UIBuildRoom(ui)
   self:makeTooltip(_S.tooltip.build_room_window.cost, 160, 228, 282, 242)
 end
 
+--! Checks what rooms are now available to build
 function UIBuildRoom:updateBuildableRooms()
   local app = self.ui.app
   self.category_rooms = {}
   for i, category in ipairs({"diagnosis", "treatment", "clinics", "facilities"}) do
     local rooms = {}
     self.category_rooms[i] = rooms
-    for _, room in ipairs(app.world.available_rooms) do
+    local room_discoveries = self.ui.hospital.room_discoveries
+    for room_id, _ in pairs(room_discoveries) do
+      local room = room_discoveries[room_id].room
```

Comment:
You obtain the `room` here, even though you need it only at line 118 if lines 116 and 117 yield true.
Depending on how often that doesn't happen, you're getting `room`s for nothing.

Perhaps move the line after you decided it's going to be used? (and perhaps even delete it there by simplifying the code)